api/appendix_f.asciidoc

```diff
@@ -0,0 +1,210 @@
+
+| {CL_INVALID_IMAGE_DESCRIPTOR_anchor}
+| Returned when the specified image descriptor is `NULL` or specifies invalid values.
+
+| {CL_INVALID_IMAGE_FORMAT_DESCRIPTOR_anchor}
+| Returned when the specified image format descriptor is `NULL` or specifies invalid value.
+
+| {CL_INVALID_IMAGE_SIZE_anchor}
+| Returned when the specified image dimensions exceed the maximum dimensions for a device or all devices in a context.
+
+| {CL_INVALID_KERNEL_anchor}
+| Returned when the specified kernel is not a valid kernel.
+
+| {CL_INVALID_KERNEL_ARGS_anchor}
+| Returned when enqueing a kernel when some kernel arguments have not been set or are invalid.
+
+| {CL_INVALID_KERNEL_DEFINITION_anchor}
+| Returned when creating a kernel for multiple devices where the number of kernel arguments or kernel argument types are not the same for all devices.
+
+| {CL_INVALID_KERNEL_NAME_anchor}
+| Returned when creating a kernel when no kernel with the specified name exists in the program object.
+
+| {CL_INVALID_LINKER_OPTIONS_anchor}
+| Returned when build options passed to {clLinkProgram} are not valid.
+
+| {CL_INVALID_MEM_OBJECT_anchor}
+| Returned when a specified memory object is not a valid memory object.
+
+// This currently defined in cl.h, but it's not a core API error code.
+//| {CL_INVALID_MIP_LEVEL_anchor}
+//|
+
+| {CL_INVALID_OPERATION_anchor}
+| This is a generic error code that is returned when the requested operation is not a valid operation.
+
+| {CL_INVALID_PIPE_SIZE_anchor}
+| Returned when attempting to create a pipe with an invalid packet size or number of packets.
+
+| {CL_INVALID_PLATFORM_anchor}
+| Returned when the specified platform is not a valid platform.
+
+| {CL_INVALID_PROGRAM_anchor}
+| Returned when a specified program is not a valid program object.
+
+| {CL_INVALID_PROGRAM_EXECUTABLE_anchor}
+| Returned when the specified program is valid but has not been successfully built.
+
+| {CL_INVALID_PROPERTY_anchor}
+| Returned when a specified property name is invalid, when the value for a property name is invalid, or when the same property name is specified more than once.
+
+| {CL_INVALID_QUEUE_PROPERTIES_anchor}
+| Returned when specified queue properties are valid but are not supported by the device.
+
+| {CL_INVALID_SAMPLER_anchor}
+| Returned when a specified sampler is not a valid sampler object.
+
+| {CL_INVALID_SPEC_ID_anchor}
+| Returned when the specified specialization constant ID is not valid for the specified program.
+
+| {CL_INVALID_VALUE_anchor}
+| This is a generic error that is returned when a specified value is not a valid value.
+
+| {CL_INVALID_WORK_DIMENSION_anchor}
+| Returned by {clEnqueueNDRangeKernel} when the specified work dimension is not valid.
+
+| {CL_INVALID_WORK_GROUP_SIZE_anchor}
+| Returned by {clEnqueueNDRangeKernel} when the specified total work group size is not valid for the specified kernel or device.
+
+// This is a strange error code - replace by CL_INVALID_WORK_GROUP_SIZE?
```

Comment:
It is strange.  Probably so named to match `CL_DEVICE_MAX_WORK_ITEM_SIZES` which is also a strange name.

The error codes probably could be combined with no disruption as the entry point that can return it already can return both.  It is currently well described when to use which code though.

No action required in this PR, I'm just commenting.